Userland/Services/SystemServer/main.cpp

```diff
@@ -59,7 +59,9 @@ static void sigchld_handler(int)
             continue;
         }
 
-        if (auto result = service->did_exit(status); result.is_error())
+        if (auto result = service->did_exit(WIFEXITED(status) ? WEXITSTATUS(status) : Optional<int> {},
+                WIFSIGNALED(status) ? WTERMSIG(status) : Optional<int> {});
```

Comment:
thinking of moving that into `did_exit`. `exit_code` parameter name is somewhat a misnomer as we see you pas a `status` to be decoded—maybe going for `ErrorOr<void> Service::did_exit(int status)` would help with the logging there?